functional_programming_tests/test/with_dependency_injection.cpp

```diff
@@ -0,0 +1,528 @@
+        "set_costmap",
+        [callback](
+            std::shared_ptr<example_srvs::srv::SetMap::Request> const request,
+            std::shared_ptr<example_srvs::srv::SetMap::Response> response) {
+          callback(request, response);
+        });
+  }
+  void register_generate_path_service(
+      GeneratePathServiceCallback callback) override {
+    path_generator_ = node_->create_service<example_srvs::srv::GetPath>(
+        "generate_global_path",
+        [callback](
+            std::shared_ptr<example_srvs::srv::GetPath::Request> const request,
+            std::shared_ptr<example_srvs::srv::GetPath::Response> response) {
+          callback(request, response);
+        });
+  }
+
+  void log_info(std::string const& msg) override {
+    RCLCPP_INFO_STREAM(node_->get_logger(), msg);
+  }
+
+  void log_error(std::string const& msg) override {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), msg);
+  }
+
+ private:
+  std::shared_ptr<rclcpp::Node> node_;
+  std::shared_ptr<rclcpp::Service<example_srvs::srv::SetMap>> map_setter_;
+  std::shared_ptr<rclcpp::Service<example_srvs::srv::GetPath>> path_generator_;
+};
+
+/**
+ * @brief Convert a path from ROS message to a vector of positions
+ * @param msg The message to convert
+ * @return The path as a vector of positions
+ */
+std::vector<Position> parseGeneratedPath(
+    const std_msgs::msg::UInt8MultiArray& msg) {
+  // Check the path has an even number of elements
+  if ((msg.data.size() % 2) != 0) {
+    return {};
+  }
+  if (msg.data.size() == 0) {
+    return {};
+  }
+  std::vector<Position> path;
+
+  for (size_t idx = 0; idx < (msg.data.size() - 1); idx += 2) {
+    path.push_back({msg.data[idx], msg.data[idx + 1]});
+  }
+
+  return path;
+}
+
+/**
+ * @brief Create a sample cost map in a request message
+ * @return A shared pointer to the request message with the cost map
+ */
+std::shared_ptr<example_srvs::srv::SetMap::Request> make_costmap() {
+  auto const request = std::make_shared<example_srvs::srv::SetMap::Request>();
+
+  request->map = std_msgs::msg::UInt8MultiArray();
+
+  request->map.layout.dim.resize(3, std_msgs::msg::MultiArrayDimension());
+
+  request->map.layout.dim[0].label = "rows";
+  request->map.layout.dim[0].size = 8;
+  request->map.layout.dim[0].stride = 64;
+
+  request->map.layout.dim[1].label = "columns";
+  request->map.layout.dim[1].size = 8;
+  request->map.layout.dim[1].stride = 1;
+
+  request->map.layout.dim[2].label = "channel";
+  request->map.layout.dim[2].size = 1;
+  request->map.layout.dim[2].stride = 1;
+
+  request->map.data = {0, 0, 0, 0, 0, 0, 0, 0,  //
+                       0, 0, 0, 1, 0, 0, 0, 0,  //
+                       0, 0, 0, 1, 0, 0, 0, 0,  //
+                       0, 0, 1, 1, 1, 0, 0, 0,  //
+                       0, 0, 1, 0, 1, 1, 0, 0,  //
+                       0, 0, 1, 0, 0, 0, 0, 0,  //
+                       0, 0, 0, 0, 0, 0, 0, 0,  //
+                       0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
Are the comments to stop the linter?

Reply:
yessir. I can't find a reference right now but I did read about doing this, which is called adding "trivial comments" (at least that's what i thought it was called)